JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -0,0 +1,524 @@
+                function supportMonitoring() {
+                    var result = false;
+                    if (!extensions.IsNullOrUndefined(XMLHttpRequest)) {
+                        result = true;
+                    }
+
+                    return result;
+                }
+
+                function interceptOpen() {
+                    var originalOpen = XMLHttpRequest.prototype.open;
+                    XMLHttpRequest.prototype.open = function (method, url, async) {
+                        if (isMonitoredInstance.call(this, true)) {
+                            this.ajaxData = new ajaxRecord();
+                            attachToOnReadyStateChange.call(this);
+                            commands.TryCatchTraceWrapper.call(this, "openPrefix", function () {
+                                var self = this;
+                                self.ajaxData.method = method;
+                                self.ajaxData.requestUrl = url;
+                                self.ajaxData.requestSize += url.length;
+
+                                // If not set async defaults to true 
+                                self.ajaxData.async = extensions.IsNullOrUndefined(async) ? true : async;
+
+                            });
+                        }
+
+                        return originalOpen.apply(this, arguments);
+                    };
+                }
+
+                function interceptSend() {
+                    var originalSend = XMLHttpRequest.prototype.send;
+                    XMLHttpRequest.prototype.send = function (content) {
+                        sendPrefixInterceptor.call(this, content);
+                        return originalSend.apply(this, arguments);
+                    };
+                }
+
+                function sendPrefixInterceptor(content) {
+                    if (isMonitoredInstance.call(this)) {
+                        commands.TryCatchTraceWrapper.call(this, "sendPrefix", function () {
+                            if (!extensions.IsNullOrUndefined(content) && !extensions.IsNullOrUndefined(content.length)) {
+
+                                // http://www.w3.org/TR/XMLHttpRequest/: If the request method is a case-sensitive match for GET or HEAD act as if data is null.
+                                if (this.ajaxData.method !== "GET" && this.ajaxData.method !== "HEAD") {
+                                    this.ajaxData.requestSize += content.length;
+                                }
+                            }
+
+                            this.ajaxData.requestSentTime = dateTime.Now();
+                            this.ajaxData.loadingRequest = document.readyState === "loading";
+
+                            if (!this.ajaxData.onreadystatechangeCallbackAttached) {
+
+                                // IE 8 and below does not support xmlh.addEventListener. This the last place for the browsers that does not support addEventListenener to intercept onreadystatechange
```

Comment:
Do we support IE 8 at all? Maybe we shouldn't even intercept XmlHttpRequest for older browsers?